src/main/java/frc/robot/commands/intake/IndexOneCargo.java

```diff
@@ -0,0 +1,32 @@
+package frc.robot.commands.intake;
+
+import edu.wpi.first.wpilibj2.command.ConditionalCommand;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.Constants;
+import frc.robot.subsystems.IndexerSubsystem;
+
+/**
+ * Run the indexer's gate wheel until a cargo has been indexed, then
+ */
+public class IndexOneCargo extends ConditionalCommand {
+    public IndexOneCargo(IndexerSubsystem indexer) {
+        super(
+            new WaitCommand(0), // Don't do anything if the indexer already has a cargo
+            new SequentialCommandGroup(
+                new InstantCommand(() -> { // Turn on the indexer
+                    indexer.activateConveyor();
+                    indexer.activateGateWheelForIndexing();
+                }), 
+                new WaitUntilCommand(() -> indexer.getGateWheelSupplyCurrent() > Constants.Indexer.GATE_WHEEL_INDEXING_POWER), // Once a cargo has contacted the rollers, the current drawn by the motor should increase
```

Comment:
How consistent is this check? Would there be a situation in which we do not surpass the GATE_WHEEL_INDEXING_POWER, that would then leave us in an infinite loop correct?

Reply:
Honestly, I have no idea if current detection would work at all, but, without a beam-break sensor, I won't know if there's a better solution until the robot is working.